### ISSUE TITLE
Evaluate breaking change -- AllowSynchronousIO now false in ASP.NET Core 3

### ISSUE THREAD

**Issue by brettsam** [Azure/azure-functions-host#4876]
https://github.com/aspnet/AspNetCore/issues/7644. 

Internally I had to change the placeholder warmup function to read asynchronously, but I suspect customers will get bit by this when updating to v3.

**Comment by brettsam** [comment]
I actually reversed course on this a bit -- I left our ReadAsync calls b/c they're generally better. But I had to go through and set `AllowSynchronousIO = true` in a bunch of places to workaround json deserialization in the Http extension: https://github.com/Azure/azure-webjobs-sdk-extensions/blob/dev/src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs#L469.

Not sure if there's an efficient workaround for this without moving JSON.NET forward there. Will need investigation in the future.
